AssociatingAccountWithContactHandler.cls

```diff
@@ -0,0 +1,33 @@
+public with sharing class AssociatingAccountWithContactHandler {
+    
+    public static void ContactAccontAssociation(List<Contact> contactList)
+    {
+        Set<String> emailDomainSet= new  Set<String> ();
+        for(Contact con:contactList)
+        {
+            if(con.Email==NULL)
+            {
+                continue;
+            }
+        emailDomainSet.add(con.Email.split('@').get(1));
+        }
+        if(!emailDomainSet.IsEmpty())
+        {
+            Map<String,Id> accountContactMap = new Map<String,Id>();
+            for (Account acc: [Select Id, DomainName__c from Account where DomainName__c IN : emailDomainSet ])
```

Comment:
Update this for loop. Use contact domain and contact record map here to reduce processing here.